xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubPersistenceManager.java

```diff
@@ -58,7 +58,7 @@
 
     private static final Logger log = LoggerFactory.getLogger(PubSubPersistenceManager.class);
 
-    private static final String PERSISTENT_NODES = "SELECT serviceID, nodeID, maxItems " +
+    private static final String PERSISTENT_NODES = "SELECT DISTINCT serviceID, nodeID, maxItems " +
```

Comment:
Although it makes sense that we'd like unique rows to be returned, I'm alarmed by the fact that there's a need to add the DISTINCT keyword. Is there a scenario where the original query returns a non-distinct result set?

Reply:
I've found doubled rows in database but dont know where they came from... 
this fix indeed would not fix the cause but i will prevent errors based on multiple results...